src/query/sql/src/planner/optimizer/heuristic/prune_unused_columns.rs

```diff
@@ -314,6 +322,9 @@ impl UnusedColumnPruner {
             }
 
             RelOperator::MaterializedCte(cte) => {
+                if self.apply_lazy {
```

Comment:
Currently lazy materialization is only used when there is a single table, so seems there is no need to consider `apply_lazy` here.